sqrl-planner/src/main/java/com/datasqrl/plan/global/DAGAssembler.java

```diff
@@ -114,10 +114,9 @@ public PhysicalDAGPlan assemble(SqrlDAG dag, Set<URL> jars, Map<String, UserDefi
         errors.checkFatal(materializedTable.getPrimaryKey().isDefined(), ErrorCode.TABLE_NOT_MATERIALIZE,"Table [%s] does not have a primary key and can therefore not be materialized", materializedTable);
         Preconditions.checkArgument(materializedTable.getTimestamp().size()<=1, "Should not have multiple timestamps at this point");
         errors.checkFatal(materializedTable.getType()== TableType.STATIC || materializedTable.getTimestamp().size()==1, ErrorCode.TABLE_NOT_MATERIALIZE, "Table [%s] does not have a timestamp and can therefore not be materialized", materializedTable);
-        RelNode processedRelnode = produceWriteTree(materializedTable.getPlannedRelNode(),
-            materializedTable.getTimestamp().getOnlyCandidate());
-        OptionalInt timestampIdx = materializedTable.getType() == TableType.STATIC ? OptionalInt.empty() :
-            OptionalInt.of(materializedTable.getTimestamp().getOnlyCandidate());
+        OptionalInt timestampIdx = materializedTable.getType().hasTimestamp() ?
+            OptionalInt.of(materializedTable.getTimestamp().getOnlyCandidate()) : OptionalInt.empty();
```

Comment:
getTimestamp should probably be an optional as the return type and then mapped to getOnlyCandidate

Reply:
done